smithy-model/src/main/java/software/amazon/smithy/model/Model.java

```diff
@@ -141,46 +133,66 @@ public static ModelAssembler assembler(ClassLoader classLoader) {
     }
 
     /**
-     * Returns are trait shapes in the model as a map of {@code Shape}
-     * objects to their {@link TraitDefinition} traits.
+     * Gets the trait definition of a specific trait shape ID.
+     *
+     * @param traitId ID of the shape to get the trait definition of.
+     * @return Returns the optionally found trait definition.
+     */
+    public Optional<TraitDefinition> getTraitDefinition(ToShapeId traitId) {
+        return getShape(traitId.toShapeId()).flatMap(shape -> shape.getTrait(TraitDefinition.class));
+    }
+
+    /**
+     * Gets a set of shapes in the model marked with a specific trait.
      *
-     * @return Returns all trait definitions in the model.
+     * @param trait Trait shape ID to look for on shapes.
+     * @return Returns the immutable set of matching shapes.
      */
-    public Map<Shape, TraitDefinition> getTraitDefinitions() {
```

Comment:
The public API migrations in this file need a changelog entry: `getTraitDefinitions` and `getTraitShapes`.